### ISSUE TITLE
[Bug]: User pools new attribute not detected

### ISSUE THREAD

**Issue by npenin** [hashicorp/terraform-provider-aws#36499]
### Terraform Core Version

1.6.6

### AWS Provider Version

5.41

### Affected Resource(s)

* aws_user_pool

### Expected Behavior

When adding new attributes to a user pool, the user pool should have new attributes created.

### Actual Behavior

No changes are detected for the user pool

### Terraform Configuration Files

```
resource "aws_cognito_user_pool" "pool" {
  name                = "XXX"
  deletion_protection = "ACTIVE"
  mfa_configuration   = "OPTIONAL"
  account_recovery_setting {
    recovery_mechanism {
      name     = "verified_email"
      priority = 1
    }
    recovery_mechanism {
      name     = "verified_phone_number"
      priority = 2
    }
  }

  auto_verified_attributes = ["email"]

  username_attributes = ["email", "phone_number"]

  admin_create_user_config {
    allow_admin_create_user_only = false
  }

  device_configuration {
    challenge_required_on_new_device      = true
    device_only_remembered_on_user_prompt = true
  }

  email_configuration {
    email_sending_account = "COGNITO_DEFAULT"
  }

  password_policy {
    minimum_length                   = 6
    require_lowercase                = false
    require_numbers                  = false
    require_symbols                  = false
    require_uppercase                = false
    temporary_password_validity_days = 7
  }

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "access"
    required                 = false

    string_attribute_constraints {}

  }

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "profileId"
    required                 = false

    string_attribute_constraints {}

  }
  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "email"
    required                 = true

    string_attribute_constraints {
      max_length = "2048"
      min_length = "0"
    }
  }
  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "family_name"
    required                 = true

    string_attribute_constraints {
      max_length = "2048"
      min_length = "0"
    }
  }
  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "given_name"
    required                 = true

    string_attribute_constraints {
      max_length = "2048"
      min_length = "0"
    }
  }

  software_token_mfa_configuration {
    enabled = true
  }

  user_attribute_update_settings {
    attributes_require_verification_before_update = [
      "email",
    ]
  }

  username_configuration {
    case_sensitive = false
  }

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "license_verified"
    required                 = false

  }

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "identity_verified"
    required                 = false

  }

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "hidden_cards"
    required                 = false

  }

  lifecycle {
    ignore_changes = [
      schema
    ]
  }

}
```

### Steps to Reproduce

add the following attributes:
```

  schema {
    attribute_data_type      = "String"
    developer_only_attribute = false
    mutable                  = true
    name                     = "crmId"
    required                 = false
    
    string_attribute_constraints {
      max_length = "64"
      min_length = "0"
    }
  }

  schema {
    attribute_data_type      = "Boolean"
    developer_only_attribute = false
    mutable                  = true
    name                     = "importedFromCRM"
    required                 = false
  }
```

### Debug Output

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

### Would you like to implement a fix?

None

**Comment by acwwat** [comment]
@npenin You have the following in your configuration, that's why it's ignoring any changes with `schema` blocks after creation:

```
  lifecycle {
    ignore_changes = [
      schema
    ]
  }
```

Although removing the meta-argument would lead you to the diff problem described in #21654. It seems that `string_attribute_constraints` block is not ignored from the diff, and attempting to apply the differences would fail because schema items are immutable due to https://repost.aws/knowledge-center/cognito-change-user-pool-attributes.

The diff issue was seemingly addressed by #32445 but it doesn't seem to be working, so I am not sure if there is a regression. This will need to be investigated further.

**Comment by justinretzolk** [comment]
Hey @npenin 👋 Thank you for taking the time to raise this! As mentioned above, the changes aren't being detected in your case due to the `lifecycle` block. With that in mind, I'm going to close this issue. If you run into any further issues (particularly if you wind up running into the one described above), we'd love it if you'd let us know.
